projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -22,87 +37,149 @@ export class PendingRequestComponent implements OnInit, OnChanges {
     { field: "at", order: -1 },
     { field: "member_id", order: 1 },
   ];
-  prevSortMeta = "-at member_id";
-
-  @Output() onAccept = new EventEmitter<any>();
-  @Output() onReject = new EventEmitter<any>();
-
-  groupSwitch = [
-    {
-      label: "This group only",
-    },
-    {
-      label: "All subgroups",
-    },
-  ];
+  prevSortMeta: string = "-at member_id";
+
+  requestLoading: boolean = false;
+  requestAction: string = "";
+  selection = [];
+
+  reloadData(sortBy = []) {
+    this.selection = [];
+    if (sortBy.length === 0) {
+      sortBy = this.prevSortMeta.split(" ");
+    }
+    this.groupService
+      .getManagedRequests(this.id, sortBy)
+      .subscribe((reqs: PendingRequest[]) => {
+        this.requests = [];
```

Comment:
why initializing the var to empty first?

Reply:
fixed. This is not necessary.